src/main/java/com/shapesecurity/shift/parser/Parser.java

```diff
@@ -81,11 +85,27 @@ private void consumeSemicolon() throws JsError {
 
     @NotNull
     protected <T extends Node> T markLocation(@NotNull SourceLocation startLocation, @NotNull T node) {
-        // TODO: actually mark location
-//    node.loc = Maybe.just(new SourceSpan(Maybe.nothing(), startLocation, new SourceLocation(this.lastLine+1, this.lastIndex-this.lastLineStart, this.lastIndex)));
+        locations.put(
+                node,
+                Maybe.just(
+                        new SourceSpan(
+                                Maybe.nothing(),
+                                startLocation,
+                                new SourceLocation(
+                                        this.lastLine + 1,
+                                        this.lastIndex - this.lastLineStart,
+                                        this.lastIndex
+                                )
+                        )
+                )
+        );
         return node;
     }
 
+    public Maybe<SourceSpan> getLocation(@NotNull Node node) {
```

Comment:
Using the `HashTable` as I suggest above guarantees that this will always return a non-`null` value.